### PR TITLE
Feat/frontend search

### LINKED ISSUE #247: Feat: Frontend Search
Making sure searching works as expected on the frontend

### PR DESCRIPTION
**Pull request by AlenGanopolsky**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/247)
Added fuzzy search functionality in the frontend - currently using text placeholders since SearchCard needs to be updated to support Event data instead of Event Occurence data

# How Has This Been Tested?
Manual testing

## Screenshots
https://github.com/user-attachments/assets/e795c06d-95b2-42a9-929f-15c0a0b842e7

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: frontend/apps/mobile/contexts/auth-context.tsx [hunk 1]
```diff
@@ -158,6 +158,7 @@ export function AuthProvider({ children }: { children: ReactNode }) {
       },
       {
         onSuccess: async (resp: signupGuardianResponse) => {
+          console.log("success!");
```
Comment (adescoteaux1) [inline frontend/apps/mobile/contexts/auth-context.tsx]: remove

File: frontend/apps/web/Dockerfile [hunk 2]
```diff
@@ -12,8 +12,7 @@ COPY scripts ./scripts
 RUN bun install
 # Build the web app
 WORKDIR /app/apps/web
-
-RUN bun run build
+RUN bun x tsc -b && bun x vite build
```
Comment (adescoteaux1) [inline frontend/apps/web/Dockerfile]: undo

**Review by adescoteaux1 — APPROVED** [review]
